### PR TITLE
Process Action: Retrieval Tags

### PR DESCRIPTION
**Pull request by spolu**
## Description

The goal of this PR is to add support for tags filtering in the process action. Two passes were possible:
- (i) Add a `filterTags` object to the `ProcessActionConfigurationType` (chosen by this PR)
- (ii) Leverage the DataSourceConfigurationType's tag filtering and build UI in the DataSource selector to add tags.

(ii) is probematic because `core` does not support per data source tags (tags are instead aggregated together and applied to all data source in the current code). This has no impact since tags in DataSourceConfiguration are never used and the tagsIn and tagsNotIn columns are empty in production.

Most of the time process action will process "uniform" data coming from one data source so having global tags should work most of the time. The best situation in the end would be to have per data source tag support with `core` in_map support for tags as it has for parents and add support for tags in the data source selector which is doable for most connections but will be a challenge UX-wise for folders (you select multiple folders but tags should be applied to each folders). We can revisit that in the future and the current approach is the minimal path of resistance towards having tags support for the process action which will allow us to move all of our highlights assistants to it.

As a follow-up PR I will remove support for DataSourceConfiguration tagsIn/tagsNotIn since it's not used in production and would be counter intuitive if it was used see: https://github.com/dust-tt/dust/blob/main/front/lib/api/assistant/actions/retrieval.ts#L620

- Adds a retrievalTags state to the process action
- Add support in the Process Action Assistant Builder UI

Screenshot from 2024-05-06 16-53-14

Note: this is a prototype and I'm happy to iterate on it once we start using it or pushing it to users.

## Risk

N/A Process action is not released to anyone

## Deploy Plan

- run init_db
- deploy `front`

### CODE REVIEW

File: front/components/assistant_builder/actions/ProcessAction.tsx [hunk 1]
```diff
@@ -48,11 +49,26 @@ export function isActionProcessValid(
...
-  );
+  if (Object.keys(action.configuration.dataSourceConfigurations).length === 0) {
+    return false;
+  }
+  if (!action.configuration.timeFrame.value) {
+    return false;
+  }
+  if (
+    action.configuration.tagsFilter &&
+    action.configuration.tagsFilter.in.filter((tag) => tag === "").length > 0
```
Comment (fontanierh) [inline front/components/assistant_builder/actions/ProcessAction.tsx:60]: nit: would be slightly more readable as `action.configuration.tagsFilter.in.some(...)`

**Review by fontanierh — APPROVED** [review]
Looks great

File: front/components/assistant_builder/actions/ProcessAction.tsx (hunk 1 above)
Comment (spolu) [reply front/components/assistant_builder/actions/ProcessAction.tsx:60]: sure will update :+1:

**Comment by PopDaph** [comment]
Should we refactor all that as part of the Tags As Filter project?

**Comment by spolu** [comment]
Yes we very likely want to align UI and mechanisms 👍👍

**Comment by PopDaph** [comment]
Task created 👍🏻  https://github.com/dust-tt/tasks/issues/2065

**Pull request merged**